static/panes/executor.js

```diff
@@ -0,0 +1,780 @@
+    this.setCompilationOptionsPopover(result.compilationOptions ? result.compilationOptions.join(' ') : '');
+
+    if (this.nextRequest) {
+        var next = this.nextRequest;
+        this.nextRequest = null;
+        this.sendCompile(next);
+    }
+};
+
+
+Executor.prototype.onEditorChange = function (editor, source, langId, compilerId) {
+    if (editor === this.sourceEditorId && langId === this.currentLangId &&
+        (compilerId === undefined || compilerId === this.id)) {
+        this.source = source;
+        if (this.settings.compileOnChange) {
+            this.compile();
+        }
+    }
+};
+
+Executor.prototype.initButtons = function (state) {
+    this.filters = new Toggles(this.domRoot.find('.filters'), state.filters);
+
+    this.compileClearCache = this.domRoot.find('.clear-cache');
+
+
+    this.optionsField = this.domRoot.find('.compilation-options');
+    this.execArgsField = this.domRoot.find('.execution-arguments');
+    this.prependOptions = this.domRoot.find('.prepend-options');
+    this.fullCompilerName = this.domRoot.find('.full-compiler-name');
+    this.setCompilationOptionsPopover(this.compiler ? this.compiler.options : null);
+
+    this.compileTimeLabel = this.domRoot.find('.compile-time');
+    this.libsButton = this.domRoot.find('.btn.show-libs');
+
+    // Dismiss on any click that isn't either in the opening element, inside
+    // the popover or on any alert
+    $(document).on('mouseup', _.bind(function (e) {
+        var target = $(e.target);
+        if (!target.is(this.prependOptions) && this.prependOptions.has(target).length === 0 &&
+            target.closest('.popover').length === 0)
+            this.prependOptions.popover("hide");
+
+        if (!target.is(this.fullCompilerName) && this.fullCompilerName.has(target).length === 0 &&
+            target.closest('.popover').length === 0)
+            this.fullCompilerName.popover("hide");
+    }, this));
+
+    this.optionsField.val(this.options);
+    this.execArgsField.val(this.executionArguments);
+
+    this.shortCompilerName = this.domRoot.find('.short-compiler-name');
+    this.compilerPicker = this.domRoot.find('.compiler-picker');
+    this.setCompilerVersionPopover('');
+
+    this.topBar = this.domRoot.find('.top-bar');
+    this.bottomBar = this.domRoot.find('.bottom-bar');
+    this.statusLabel = this.domRoot.find('.status-text');
+
+    this.hideable = this.domRoot.find('.hideable');
+    this.statusIcon = this.domRoot.find('.status-icon');
+
+    this.initPanerButtons();
+};
+
+Executor.prototype.onLibsChanged = function () {
+    this.saveState();
+    this.compile();
+};
+
+Executor.prototype.initLibraries = function (state) {
+    this.libsWidget = new Libraries.Widget(this.currentLangId, this.libsButton,
+        state, _.bind(this.onLibsChanged, this));
+};
+
+Executor.prototype.updateButtons = function () {
+    if (!this.compiler) return;
+};
+
+Executor.prototype.handlePopularArgumentsResult = function (result) {
```

Comment:
not needed here (if we do, it probably needs to be refactored into a new js file)